lib/active_merchant/billing/gateways/ebanx.rb

```diff
@@ -0,0 +1,234 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class EbanxGateway < Gateway
+      self.test_url = 'https://sandbox.ebanx.com/ws/'
+      self.live_url = 'https://api.ebanx.com/ws/'
+
+      self.supported_countries = ['BR', 'CL', 'MX', 'CO', 'PE']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover, :diners_club]
+
+      self.homepage_url = 'http://www.ebanx.com/'
+      self.display_name = 'Ebanx'
+
+      CARD_BRAND = {
+        visa: "visa",
+        master: "master_card",
+        american_express: "amex",
+        discover: "discover",
+        diners_club: "diners"
+      }
+
+      URL_MAP = {
+        purchase: "direct",
+        authorize: "direct",
+        capture: "capture",
+        refund: "refund",
+        void: "cancel"
+      }
+
+      HTTP_METHOD = {
+        purchase: :post,
+        authorize: :post,
+        capture: :get,
+        refund: :post,
+        void: :get
+      }
+
+      def initialize(options={})
+        requires!(options, :integration_key)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = { payment: {} }
+        add_integration_key(post)
+        add_operation(post)
+        add_invoice(post, money, options)
+        add_customer_data(post, payment, options)
+        add_payment(post, payment)
+        add_address(post, options)
+
+        commit(:purchase, post)
+      end
+
+      def authorize(money, payment, options={})
+        post = { payment: {} }
+        add_integration_key(post)
+        add_operation(post)
+        add_invoice(post, money, options)
+        add_customer_data(post, payment, options)
+        add_payment(post, payment)
+        add_address(post, options)
+        post[:payment][:creditcard][:auto_capture] = false
+
+        commit(:authorize, post)
+      end
+
+      def capture(money, authorization, options={})
+        post = {}
+        add_integration_key(post)
+        post[:hash] = authorization
+        post[:amount] = amount(money)
```

Comment:
So, I went back and forth on this one. On the one hand, I prefer to have them in methods if they're used more than once. But, on the other hand, I wondered if that was more confusing cause for `purchase` and `authorize` the field name is `amount_total` and it's added in the `add_invoice` method.

This felt a little more explicit to me to say that we're adding this exact `amount` field rather than having amounts being added in two different methods.